core/chainmgr/chainmgr.go

```diff
@@ -196,7 +196,7 @@ func (self *ChainManager) initMainLedger(stateHashHeight uint32) error {
 }
 
 func (self *ChainManager) initShardLedger(shardInfo *ShardInfo) error {
-	if shardInfo != nil && ledger.GetShardLedger(shardInfo.ShardID) != nil {
+	if ledger.GetShardLedger(shardInfo.ShardID) != nil {
```

Comment:
necessary change here?

Reply:
If we don't remove it, we need to judge ```shardInfo == nil ``` in addition